test/e2e/utils/test_context.go

```diff
@@ -232,9 +232,12 @@ func (t *TestContext) Tidy() error {
 // Destroy is for cleaning up the docker images for testing
 func (t *TestContext) Destroy() {
 	//nolint:gosec
-	cmd := exec.Command("docker", "rmi", "-f", t.ImageName)
-	if _, err := t.Run(cmd); err != nil {
-		warnError(err)
+	// if image name is not present or not provided skip execution of docker command
+	if t.ImageName != "" {
+		cmd := exec.Command("docker", "rmi", "-f", t.ImageName)
+		if _, err := t.Run(cmd); err != nil {
+			warnError(err)
+		}
```

Comment:
Looks good! Seems like when `t.ImageName` is set by hardcoding the value in some of the places while creating a test context, like this: https://github.com/kubernetes-sigs/kubebuilder/blob/a5b246f1ff5b6444ef75bb82c73ae557f47b2438/test/e2e/utils/test_context.go#L91.

So let's also add a warning when `t.ImageName` is empty and we don't execute the command. Something like this:

```suggestion
	if t.ImageName != "" {
		cmd := exec.Command("docker", "rmi", "-f", t.ImageName)
		if _, err := t.Run(cmd); err != nil {
			warnError(err)
		}
	} else {
	           warnErr(fmt.Errorf("ImageName not set, skipping cleaning up of docker image"))
	}	
```

Wdyt? 
cc: @camilamacedo86 ?

Reply:
if there is no image we are already giving a warning error then why adding else.